src/library.js

```diff
@@ -2127,7 +2127,9 @@ LibraryManager.library = {
 
   _arraySum: function(array, index) {
     var sum = 0;
-    for (var i = 0; i <= index; sum += array[i++]);
+    for (var i = 0; i <= index; sum += array[i++]) {
+      // no-op
+    }
```

Comment:
Closure barks at `for(...;...;...);` thinking the `;` is accidental.

Reply:
That would be one byte longer, and micro-slower as `i` would get referenced twice? :/

Reply:
Does closure not do such micro-optimization at the cost of readability? I think the readability cost here is annoying enough we should rely on closure.

Reply:
Closure does transform the code back to `for (var i = 0; i <= index; sum += array[i++]);`, it just does not want to see that form in the input.

Reply:
Sorry if I wasn't clear - I meant, can we write the human-readable form (with `for (..) { sum += array[i] }`) and rely on closure to minify it to the optimal form? If so then I think writing the human-readable form is the best thing.

Reply:
Oh gotcha - unfortunately it does not look like Closure is able to do such a transform (https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%252F%252F%2520ADD%2520YOUR%2520CODE%2520HERE%250Afunction%2520sumUp(arr)%2520%257B%250A%2520%2520var%2520sum%2520%253D%25200%253B%250A%2520%2520var%2520index%2520%253D%2520arr.length-1%253B%250A%2520%2520for%2520(var%2520i%2520%253D%25200%253B%2520i%2520%253C%253D%2520index%253B%2520i%252B%252B)%2520%257B%250A%2520%2520%2520%2520sum%2520%252B%253D%2520array%255Bi%255D%253B%250A%2520%2520%257D%250A%2520%2520return%2520sum%253B%250A%257D%250Aconsole.log(sumUp(new%2520Array%255B10%255D))%253B%250A%250A)

Reply:
That's too bad. I guess the less-readable form is ok then, especially since it looks like no one aside from me cares much about it.